src/validate_oNHM_daily_output.R

```diff
@@ -0,0 +1,42 @@
+library(assertthat)
+library(ncmeta)
+
+validate_oNHM_daily_output <- function(var, fn, test_date, data_nc, hruids, time, time_fixed, n_hrus = 109951) {
+  
+  ##### Test: NetCDF dimensions and variables are as expected #####
+  
+  # Get a bit more detailed metadata
+  meta_info <- nc_meta(fn)
+  atts_var_i <- which(meta_info$attribute$variable == var)
+  atts_units_i <- which(meta_info$attribute$name == "units")
+  var_unit_i <- atts_units_i[atts_units_i %in% atts_var_i]
+  
+  assert_that(meta_info$dimension$name[1] == "hruid")
+  assert_that(meta_info$dimension$name[2] == "time")
+  assert_that(meta_info$dimension$length[1] == n_hrus) # Expect all hruids
+  assert_that(meta_info$dimension$length[2] == 1) # Expect only one date
+  
+  # Test unit for variable
+  assert_that(meta_info$attribute$value[[var_unit_i]] == "mm") # Expect millimeters
+  
+  ##### Test: NetCDF hruids are in expected order and the expected data type #####
+  assert_that(is.integer(hruids))
+  # need to make hruids a vector instead of matrix w 1 dim in order to test sameness
+  assert_that(all(as.vector(hruids) == 1:n_hrus)) 
```

Comment:
If they ever change the HRU numbering scheme so they aren't consecutive this will need to be changed, but we don't need to worry about that now

Reply:
well yes, but that's the idea because if they change that and they are no longer in the same order, then that could be a problem!